apps/schema/src/components/TableEditModal.vue

```diff
@@ -1,156 +1,194 @@
+        id="table_name"
+        v-model="table.name"
+        label="Name"
+        :errorMessage="nameInvalid"
+      />
+      <InputText
+        id="table_description"
+        v-model="table.description"
+        label="Description"
+      />
+      <InputSelect
+        v-if="rootTable !== undefined"
+        id="table_extends"
+        v-model="table.inherit"
+        :required="true"
+        :options="inheritOptions"
+        :readonly="table.oldName !== undefined"
+        :errorMessage="subclassInvalid"
+        label="Extends table (can not be edited after creation)"
+      />
+      <InputString
+        id="table_semantics"
+        :list="true"
+        v-model="table.semantics"
+        label="Semantics (comma separated list of IRI defining type, and/or keyword 'id')"
+      />
     </template>
     <template v-slot:footer>
-      <ButtonAction @click="$emit('close')">Close</ButtonAction>
-    </template>
-  </LayoutModal>
-  <!-- alter or add a table -->
-  <LayoutModal v-else :title="title" :show="true" @close="$emit('close')">
-    <template v-slot:body>
-      <LayoutForm v-if="(tableDraft.oldName = null)">
-        <InputString v-model="tableDraft.name" label="Table name" />
-        <InputText v-model="tableDraft.description" label="Table Description" />
-        <InputString
-          :list="true"
-          v-model="tableDraft.semantics"
-          label="semantics (comma separated list of IRI defining type, and/or keyword 'id')"
-        />
-      </LayoutForm>
-      <div v-else>ALTER IS NOT YET IMPLEMENTED</div>
-    </template>
-    <template v-slot:footer>
-      <MessageSuccess v-if="success">{{ success }}</MessageSuccess>
-      <MessageError v-if="graphqlError">{{ graphqlError }}</MessageError>
-      <ButtonAlt @click="$emit('close')">Close</ButtonAlt>
-      <ButtonAction v-if="tableDraft.oldName == false" @click="executeCommand"
-        >{{ action }}
-      </ButtonAction>
+      <ButtonAction @click="close" :disabled="isDisabled">Apply</ButtonAction>
+      <ButtonAlt @click="cancel">Cancel</ButtonAlt>
     </template>
   </LayoutModal>
 </template>
 
 <script>
-import {request} from 'graphql-request';
-
 import {
-  ButtonAction,
-  ButtonAlt,
   InputString,
   InputText,
-  LayoutForm,
   LayoutModal,
-  MessageError,
-  MessageSuccess,
-  SigninForm,
-  Spinner
-} from '@mswertz/emx2-styleguide';
+  IconAction,
+  ButtonAction,
+  MessageWarning,
+  InputSelect,
+  ButtonAlt,
+} from "molgenis-components";
 
 export default {
   components: {
-    MessageSuccess,
-    MessageError,
-    ButtonAction,
-    ButtonAlt,
     LayoutModal,
     InputString,
     InputText,
-    LayoutForm,
-    Spinner,
-    SigninForm
+    IconAction,
+    ButtonAction,
+    MessageWarning,
+    InputSelect,
+    ButtonAlt,
   },
   props: {
-    schema: String,
-    table: {type: Object, default: () => {}}
+    /** Table metadata object entered as v-model, for updates */
+    value: {
+      type: Object,
```

Comment:
required ?

Reply:
updated comment. In case of existing table this is required, otherwise optional